injected/src/features/message-bridge.js

```diff
@@ -0,0 +1,168 @@
+            console.log(`[isolated:incoming] ${InstallProxy.NAME}`, e.detail);
+            const incoming = InstallProxy.create(e.detail);
+            if (!incoming) return;
+            this.installProxyFor(incoming.featureName, args.messagingConfig);
+        });
+        window.addEventListener(appendToken(ProxyNotification.NAME), (/** @type {CustomEvent<unknown>} */ e) => {
+            console.log(`[isolated:incoming] ${ProxyNotification.NAME}`, e.detail);
+            const notification = ProxyNotification.create(e.detail);
+            if (!notification) return;
+            this.proxyNotification(notification);
+        });
+        window.addEventListener(appendToken(ProxyRequest.NAME), (/** @type {CustomEvent<unknown>} */ e) => {
+            console.log(`[isolated:incoming] ${ProxyRequest.NAME}`, e.detail);
+            const request = ProxyRequest.create(e.detail);
+            if (!request) return;
+            this.proxyRequest(request, appendToken);
+        });
+        window.addEventListener(appendToken(SubscriptionRequest.NAME), (/** @type {CustomEvent<unknown>} */ e) => {
+            console.log(`[isolated:incoming] ${SubscriptionRequest.NAME}`, e.detail);
+            const subscription = SubscriptionRequest.create(e.detail);
+            if (!subscription) return;
+            this.proxySubscription(subscription, appendToken);
+        });
+        window.addEventListener(appendToken(SubscriptionUnsubscribe.NAME), (/** @type {CustomEvent<unknown>} */ e) => {
+            console.log(`[isolated:incoming] ${SubscriptionUnsubscribe.NAME}`, e.detail);
+            const unsubscribe = SubscriptionUnsubscribe.create(e.detail);
+            if (!unsubscribe) return;
+            this.removeSubscription(unsubscribe);
+        });
+    }
+
+    /**
+     * @param {string} featureName
+     * @param {import('@duckduckgo/messaging').MessagingConfig} config
+     */
+    installProxyFor(featureName, config) {
+        if (this.proxies.has(featureName)) return;
+        const allowed = this.getFeatureSetting('allowed');
+        if (!allowed.includes(featureName)) return;
+
+        const ctx = { ...this.messaging.messagingContext, featureName };
+        const messaging = new Messaging(ctx, config);
+        this.proxies.set(featureName, messaging);
+        console.log('[isolated] did install proxy for ', featureName);
+    }
+
+    /**
+     * @param {ProxyRequest} request
+     * @param {(s: string) => string} appendToken
+     */
+    proxyRequest(request, appendToken) {
+        const { id, featureName, method, params } = request;
+
+        const proxy = this.proxies.get(featureName);
+        if (!proxy) return console.warn('[isolated] proxy was not installed');
+
+        const responseName = appendToken(ProxyResponse.NAME + '-' + id);
+        console.log('[isolated] will proxy', request);
+
+        proxy
+            ?.request(method, params)
+            // eslint-disable-next-line promise/prefer-await-to-then
```

Comment:
Can you make the method async?

Reply:
I kinda like that it's not async. It's conceptually "pass on and return immediately", and we're not using the result anyways. If we need it in the future, we can easily add a result.